src/game.cpp

```diff
@@ -10459,6 +10459,23 @@ bool game::plmove( int dx, int dy, int dz )
         return true;
     }
 
+    //Wooden Fence Gate: vault over it if we are running, open it if we are walking
+    if( m.ter( dest_loc ) == t_fencegate_c ) {
+        if( u.move_mode == "run" ) {
+            add_msg( _( "You vault over the %1$s." ), m.ter( dest_loc )->name().c_str() );
+            //do nothing here, let the regular code handle the movement
+        } else if( u.move_mode == "walk" ) {
+            add_msg( _( "You open the %1$s." ), m.ter( dest_loc )->name().c_str() );
+            m.open_door( dest_loc, !m.is_outside( u.pos() ) );
```

Comment:
You're not handling the case where opening the door fails.

Reply:
Are there any existing conditions for this case yet, or are you suggesting this for consistency with possible future changes to the codebase?

Reply:
I checked, and opening can only fail for things with the OPENCLOSE_INSIDE flag, which is obviously not present for `t_fencegate_c`, but could be added by a mod (e.g. for a padlocked gate).

Reply:
Yes you have a point there.